airflow/providers/sftp/hooks/sftp.py

```diff
@@ -120,12 +120,11 @@ def get_conn(self) -> pysftp.Connection:
 
     def close_conn(self) -> None:
         """
-        Closes the connection. An error will occur if the
-        connection wasnt ever opened.
+        Closes the connection
         """
-        conn = self.conn
-        conn.close()  # type: ignore
-        self.conn = None
+        if self.conn is not None:
```

Comment:
```suggestion
        if not self.conn:
```
Maybe better

Reply:
Good suggestion. I think `is not None` is more clear and cause less error.
like this 
```python
>>> x = 0
>>> x is not None
True
>>> not x
True
>>> y = 1
>>> y is not None
True
>>> not y
False
>>> z = None
>>> z is not None
False
>>> not z
True
```
check this: https://stackoverflow.com/questions/2710940/python-if-x-is-not-none-or-if-not-x-is-none

Reply:
I like "is not None" a bit better in this case. it's a little longer, but I think we should optimise for readability and "is not None" reads better in this case.